lib/kennel/parts_serializer.rb

```diff
@@ -26,31 +22,36 @@ def write_changed(parts)
       used = []
 
       Utils.parallel(parts, max: 2) do |part|
-        path = "generated/#{part.tracking_id.tr("/", ":").sub(":", "/")}.json"
+        path = path_for_tracking_id(part.tracking_id)
 
-        used << File.dirname(path) # only 1 level of sub folders, so this is enough
+        used << File.dirname(path) # we have 1 level of sub folders, so this is enough
         used << path
 
-        payload = part.as_json.merge(api_resource: part.class.api_resource)
-        write_file_if_necessary(path, JSON.pretty_generate(payload) << "\n")
+        content = part.as_json.merge(api_resource: part.class.api_resource)
+        write_file_if_necessary(path, content)
       end
 
       used
     end
 
-    def directories_to_clean_up
-      if filter.project_filter
-        filter.project_filter.map { |project| "generated/#{project}" }
+    def existing_paths
```

Comment:
I'm not a fan of this change (which got merged already :-( ) because it's asymmetric and misleading. The first two branches only return files, not directories; the third returns both. Does that difference matter? It's not clear. The method name doesn't make that clear either.

Reply:
ah I thought your only concern was refactoring the paths even more :(

Reply:
it returns "files or folders" 🤷 
_paths is kinda vague for that reason
`_files_or_folders` is kinda ugly

Reply:
https://github.com/grosser/kennel/pull/241